specs/lib/updateVersionSpec.js

```diff
@@ -0,0 +1,265 @@
+                        technique: 'technique',
+                        values: {
+                            ambient: [0.0, 0.0, 0.0, 1.0]
+                        }
+                    }
+                }
+            },
+            meshes: {
+                mesh: {
+                    primitives: [
+                        {
+                            primitive: WebGLConstants.TRIANGLES
+                        }
+                    ]
+                }
+            },
+            nodes: {
+                node: {
+                    rotation: [0.0, 0.0, 1.0, 0.0],
+                    instanceSkin: {
+                        skeletons: ['skeleton'],
+                        skin: 'skin',
+                        meshes: ['mesh']
+                    }
+                }
+            },
+            techniques: {
+                technique: {
+                    pass: 'pass',
+                    passes: {
+                        pass: {
+                            instanceProgram: {
+                                attributes: {
+                                    attribute: 'TEST_ATTRIBUTE'
+                                },
+                                program: 'program',
+                                uniforms: {
+                                    uniform: 'TEST_UNIFORM'
+                                }
+                            },
+                            states: ['TEST_STATE']
+                        }
+                    }
+                }
+            }
+        };
+        updateVersion(gltf, {
+            targetVersion: '1.0'
+        });
+        expect(gltf.asset.version).toBe('1.0');
+        expect(gltf.asset.profile).toEqual({});
+        expect(gltf.version).not.toBeDefined();
+        expect(gltf.extensionsUsed).toEqual(['extension']);
+        var material = gltf.materials.material;
+        expect(material.technique).toBe('technique');
+        expect(material.values).toEqual({
+            'ambient': [0.0, 0.0, 0.0, 1.0]
+        });
+        var primitive = gltf.meshes.mesh.primitives[0];
+        expect(primitive.primitive).not.toBeDefined();
+        expect(primitive.mode).toBe(WebGLConstants.TRIANGLES);
+        var node = gltf.nodes.node;
+        expect(node.rotation).toEqual([0.0, 0.0, 0.0, 1.0]);
+        expect(node.skeletons).toEqual(['skeleton']);
+        expect(node.skin).toBe('skin');
+        expect(node.meshes).toEqual(['mesh']);
+        var technique = gltf.techniques.technique;
+        expect(technique.pass).not.toBeDefined();
+        expect(technique.passes).not.toBeDefined();
+        expect(technique.attributes).toEqual({
+            attribute: 'TEST_ATTRIBUTE'
+        });
+        expect(technique.program).toBe('program');
+        expect(technique.uniforms).toEqual({
+            uniform: 'TEST_UNIFORM'
+        });
+        expect(technique.states).toEqual(['TEST_STATE']);
+    });
+
+    it('updates glTF from 1.0 to 1.1', function() {
```

Comment:
> glExtensionsUsed and extensionsRequired

Those properties have no analogue in older versions of glTF; they are just new properties. I can't take a glTF1.0 `extensionsUsed` and make it `extensionsRequired` because we don't know which ones were just used vs. required. As such, I don't think there is any safe updating that can be done here.

> buffer.type removal

Thank you, I did miss that one

> technique.functions.scissor and technique.states.enable = 3089 removal

Agreed, I will add this.

>  Adds all properties that were formerly not required, e.g., asset, buffer.byteLength, bufferView.byteLength, etc.

I'll do `byteLength`, but adding the asset if it doesn't exist is handled by `addDefaults`.

>  technique.states.functions.blendColor / technique.states.functions.depthRange parameter changes.

I think this is in the same realm as camera.perspective properties. If someone has a property that is set to an invalid value, I don't think that it is safe to just change the value to a valid one. Even though it was never explicitly forbidden in 1.0, those restrictions were still imposed by WebGL. I can maybe log a warning or something, but I don't think we can just change the value.